### PR TITLE
Adding additional details for debugging about registry pushes

### PR DESCRIPTION
**Pull request by dgershman**
## Description

This enables some additional logging information for pushes to registries.

## Related Issue

Relates to #1218 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: src/internal/packager/images/push.go [hunk 1]
```diff
@@ -52,6 +53,11 @@ func (i *ImgConfig) PushToZarfRegistry() error {
 	spinner := message.NewProgressSpinner("Storing images in the zarf registry")
 	defer spinner.Stop()
 
+	if message.GetLogLevel() >= message.DebugLevel {
+		logs.Warn.SetOutput(spinner)
```
Comment (Racer159) [inline src/internal/packager/images/push.go:57]:
```suggestion
	if message.GetLogLevel() >= message.DebugLevel {
		spinner.EnablePreserveWrites()
		logs.Warn.SetOutput(spinner)
```

If possible, can you also add this line in `src/internal/packager/images/pull.go` as well?  Looking at your code made me realize in many scenarios we are discarding this information when the spinner is destroyed.

**Review by Racer159 — CHANGES_REQUESTED** [review]
Looks good overall, and thanks for the PR!  Requesting this line be added as well since it won't destroy the messages when the spinner completes (if it completes).

Also do you have any log snippets from when you saw the infinite spin with this in place?  (was testing locally over this past weekend and while I could significantly slow the Zarf install I never saw it actually infinite spin myself yet).

**Comment by Racer159** [comment]
> 

Thanks, that is good to know - I was able to see the retry behavior happen in my tests by intentionally killing the connection to trigger them manually (by deleting or otherwise manipulating the registry pod) - what is odd is that is the exact opposite of what was reported by @corang (and others) https://github.com/defenseunicorns/zarf/issues/1218#issuecomment-1403703898

**Review by Racer159 — APPROVED** [review]
lgtm

**Review by YrrepNoj — APPROVED** [review]

**Comment by dgershman** [comment]
> > 
> 
> Thanks, that is good to know - I was able to see the retry behavior happen in my tests by intentionally killing the connection to trigger them manually (by deleting or otherwise manipulating the registry pod) - what is odd is that is the opposite of what was reported by @corang (and others) [#1218 (comment)](https://github.com/defenseunicorns/zarf/issues/1218#issuecomment-1403703898) - more logs will definitely help since this is likely shaping up to be a multi-faceted issue

I started a new ticket to track my findings https://github.com/defenseunicorns/zarf/issues/1568.

**Pull request merged**